### ISSUE TITLE
[BUG] CronExpression.validateSchedule() does not correctly check for interval size.

### ISSUE THREAD

**Issue by Tyler-Gelvin** [jobrunr/jobrunr#858]
### JobRunr Version

6.3.0

### JDK Version

eclipse-temurin:17-jdk

### Your SQL / NoSQL database

org.postgresql:postgresql:42.6.0

### What happened?

My use case is the cron schedule `1,31 * * * * *`.
This cron schedule fails in validateSchedule with IllegalArgumentException saying "The smallest interval for recurring jobs is 5 seconds." It should not fail, the interval between jobs is 30 seconds.

This failure happens to any cron schedule that triggers in the first 5 seconds of the year (and not to any others). Here are some other cases that should be different.
This only triggers once a year, but fails:
`1 0 0 1 1 *`

These trigger more or less every second, but still pass:
`* 1-59 * * * *`
`5-59 * * * * *`

One way to fix this would be to remove validateSchedule. Otherwise you would need to figure out what a "too frequent" interval is in the context of cron. At the extreme you would need to scan through a whole year to see if any two events are less than 5 seconds apart.

### How to reproduce?

```java
   CronExpression.create("1,31 * * * * *").validateSchedule();
```

**Comment by rdehuyss** [comment]
Thanks for this! Could you please update your Github profile as requested by the community guidelines?

**Comment by rdehuyss** [comment]
Fixed in branch 6.4
